Aspire Budgeting/Views/CardView/BaseCardView.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  BaseCardView.swift
+//  Aspire Budgeting
+//
+
+import SwiftUI
+
+struct BaseCardView<Content: View>: View {
+  let colorInfo: ColorInfo
+  var minY: CGFloat = 0
+  var curY: CGFloat = 0
+
+  private let cornerRadius: CGFloat = 24
+  private let height: CGFloat = 163
+
+  private let shadowRadius: CGFloat = 14
+  private let shadowYOffset: CGFloat = 4
+
+  private let content: Content
+
+  private var gradient: LinearGradient {
+    Color.fondGradientFrom(startColor: colorInfo.gradientStartColor,
+                           endColor: colorInfo.gradientEndColor)
+  }
+
+  private var offsetY: CGFloat {
+    curY < minY ? minY - curY : 0
+  }
+
+  init(colorInfo: BaseCardView<Content>.ColorInfo,
+       minY: CGFloat = 0,
+       curY: CGFloat = 0,
+       @ViewBuilder content: () -> Content) {
+    self.colorInfo = colorInfo
+    self.minY = minY
+    self.curY = curY
+    self.content = content()
+  }
+
+  var body: some View {
+    ZStack(alignment: .top) {
+      containerBox
+      content
+    }
+    .offset(y: offsetY)
+  }
+}
+
+extension BaseCardView {
+  private var containerBox: some View {
+    Rectangle()
+      .fill(gradient)
+      .cornerRadius(cornerRadius)
+      .frame(height: height)
+  }
+}
+
+// MARK: - Internal Types
+extension BaseCardView {
+  struct ColorInfo {
+    let gradientStartColor: Color
+    let gradientEndColor: Color
+    let shadowColor: Color
+  }
+
+  struct CardViewItem {
+    let title: String
+    let availableTotal: AspireNumber
+    let budgetedTotal: AspireNumber
+    let spentTotal: AspireNumber
+    let progressFactor: Double
+    let categories: [Category]
+  }
+}
+
+// MARK: - Previews
+struct CardView_Previews: PreviewProvider {
+  static var previews: some View {
+    Group {
+      Group {
+        BaseCardView<Text>(colorInfo: .init(gradientStartColor: .materialBrown800,
+                                  gradientEndColor: .materialBrown800,
+                                  shadowColor: .materialBrown800)) {Text("Hello")}
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/2082c16a188d4a4c547c31ed9660c16b0801ec5c/Aspire Budgeting/Views/CardView/BaseCardView.swift#L83"></span>Closure expressions should have a single space inside each brace.<br />
<code>closure_spacing</code> <code>BaseCardView.swift:83</code></td>
    </tr>
  </tbody>
</table>